snap/info_snap_yaml_test.go

```diff
@@ -39,14 +41,801 @@ type: app
 `)
 
 func (s *InfoSnapYamlTestSuite) TestSimple(c *C) {
-	info, err := InfoFromSnapYaml(mockYaml)
+	info, err := snap.InfoFromSnapYaml(mockYaml)
 	c.Assert(err, IsNil)
 	c.Assert(info.Name, Equals, "foo")
 	c.Assert(info.Version, Equals, "1.0")
-	c.Assert(info.Type, Equals, TypeApp)
+	c.Assert(info.Type, Equals, snap.TypeApp)
 }
 
 func (s *InfoSnapYamlTestSuite) TestFail(c *C) {
-	_, err := InfoFromSnapYaml([]byte("random-crap"))
+	_, err := snap.InfoFromSnapYaml([]byte("random-crap"))
 	c.Assert(err, ErrorMatches, "(?m)info failed to parse:.*")
 }
+
+type YamlSuite struct{}
+
+var _ = Suite(&YamlSuite{})
+
+func (s *YamlSuite) TestUnmarshalGarbage(c *C) {
+	_, err := snap.InfoFromSnapYaml([]byte(`"`))
+	c.Assert(err, ErrorMatches, ".*: yaml: found unexpected end of stream")
+}
+
+func (s *YamlSuite) TestUnmarshalEmpty(c *C) {
+	info, err := snap.InfoFromSnapYaml([]byte(``))
+	c.Assert(err, IsNil)
+	c.Assert(info.Plugs, HasLen, 0)
+	c.Assert(info.Slots, HasLen, 0)
+	c.Assert(info.Apps, HasLen, 0)
+}
+
+// Tests focusing on plugs
+
+func (s *YamlSuite) TestUnmarshalStandaloneImplicitPlug(c *C) {
+	// NOTE: yaml content cannot use tabs, indent the section with spaces.
```

Comment:
Any need to repeat [the YAML spec](http://www.yaml.org/spec/1.2/spec.html#Basic) here?